### PR TITLE
feat: added train-test split

### PR DESCRIPTION
**Pull request by alvgonfri**
I have modified the preprocessing to include the division between training and test sets, including the following sections:

- **Data cleaning:** unified and removed columns from the original dataset.
- **Train-test split:** after discarding irrelevant information during data cleaning, we split the cleaned dataset into training and test sets.
- **Train set preprocessing:** applied the remaining preprocessing steps (one-hot encoding, PCA, balancing) to the training set.
- **Test set preprocessing:** processed the test set in the same way as the training set, when needed.
- **Unsupervised preprocessing:** processed datasets that will be used for unsupervised learning.

### CODE REVIEW

**Review by alvarobernal2412 — APPROVED** [review]
Great job!!

**Review by adrrf — APPROVED** [review]
lgtm 😈
